redisinsight/api/src/modules/rdi/client/api.rdi.client.ts

```diff
@@ -153,4 +173,30 @@ export class ApiRdiClient extends RdiClient {
       await this.connect();
     }
   }
+
+  private async pollActionStatus(actionId: string): Promise<any> {
+    const startTime = Date.now();
+    while (true) {
+      if (Date.now() - startTime > MAX_POLLING_TIME) {
+        throw new RdiPipelineTimeoutException();
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, POLLING_INTERVAL));
```

Comment:
Is it better to sleep after action rather then before? what if everything was processed and ready in ms?

Reply:
If everything will proceed in ms it will still work. BTW i agree that better to sleep after(requests will be proceeded really quickly)
Thanks